src/components/report/ReportAdminModal.tsx

```diff
@@ -1,30 +1,35 @@
-import { Button, Form, Input, Select } from 'antd'
+import { Button, Form, Input, Radio, Select } from 'antd'
 import { useForm } from 'antd/lib/form/Form'
 import { useApp } from 'lodestar-app-element/src/contexts/AppContext'
+import { useAuth } from 'lodestar-app-element/src/contexts/AuthContext'
 import React, { useState } from 'react'
 import { useIntl } from 'react-intl'
 import { handleError } from '../../helpers'
-import AdminModal, { AdminModalProps } from '../admin/AdminModal'
-import { reportMessages } from './translations'
 import { useMutateReport, useMutateReportPermissionGroup } from '../../hooks/report'
-import PermissionGroupSelector from '../form/PermissionGroupSelector'
-import { useAuth } from 'lodestar-app-element/src/contexts/AuthContext'
 import { ReportProps } from '../../types/report'
+import AdminModal, { AdminModalProps } from '../admin/AdminModal'
+import PermissionGroupSelector from '../form/PermissionGroupSelector'
+import { reportMessages } from './translations'
 
 type FieldProps = {
   title: string
   type: string
+  formType: 'report' | 'dashboard'
   question: string
+  dashboard: string
   viewPermissions?: string[]
 }
 
 const ReportAdminModal: React.FC<
   {
     report?: ReportProps
     onRefetch?: () => void
+    reports: ReportProps[]
   } & AdminModalProps
-> = ({ report, onRefetch, onCancel, ...props }) => {
+> = ({ report, onRefetch, onCancel, reports, ...props }) => {
+  const originFormType = (report && Object.keys(report.options?.metabase?.resource)[0]) || 'question'
   const [loading, setLoading] = useState(false)
+  const [formType, setFormType] = useState<'question' | 'dashboard'>('question')
```

Comment:
replace default value 'question' with `Object.keys(report.options?.metabase?.resource)[0]) || 'question'`

Reply:
Thank for review. Modifyed.

Reply:
better to change into the following format:
```
options: {
    metabase: {
        type: "dashboard",
        no: 123
    }
}
```